experiments/ClimaEarth/components/ocean/oceananigans.jl

```diff
@@ -0,0 +1,322 @@
+            filename = joinpath(output_dir, "ocean_diagnostics.nc"),
+            schedule = OC.TimeInterval(86400),
+        )
+        ocean.output_writers[:diagnostics] = netcdf_writer
+    end
+
+    sim = OceananigansSimulation(ocean, area_fraction, ocean_properties, remappers)
+    return sim
+end
+
+###############################################################################
+### Functions required by ClimaCoupler.jl for a SurfaceModelSimulation
+###############################################################################
+
+# Timestep the simulation forward to time `t`
+Interfacer.step!(sim::OceananigansSimulation, t) = OC.time_step!(sim.sim, float(t) - sim.sim.model.clock.time)
+
+function OC.time_step!(atmos::ClimaAtmosSimulation, Δt)
+    # It is not Oceananigans job to step the model
+    return nothing
+end
+
+# We always want the surface, so we always set zero(pt.lat) for z
+@inline to_node(pt::CA.ClimaCore.Geometry.LatLongPoint) = pt.long, pt.lat, zero(pt.lat)
+# This next one is needed if we have "LevelGrid"
+@inline to_node(pt::CA.ClimaCore.Geometry.LatLongZPoint) = pt.long, pt.lat, zero(pt.lat)
+
+instantiate(L) = L()
+
+function map_interpolate(points, oc_field::OC.Field) #, loc, grid)
+    loc = map(instantiate, OC.Fields.location(oc_field))
+    grid = oc_field.grid
+    data = oc_field.data
+
+    # TODO: There has to be a better way
+    min_lat, max_lat = extrema(OC.φnodes(grid, OC.Center(), OC.Center(), OC.Center()))
+
+    map(points) do pt
+        FT = eltype(pt)
+
+        # The oceananigans grid does not cover the entire globe, so we should not
+        # interpolate outside of its latitude bounds. Instead we return 0
+        min_lat < pt.lat < max_lat || return FT(0)
+
+        fᵢ = OC.Fields.interpolate(to_node(pt), data, loc, grid)
+        convert(FT, fᵢ)::FT
+    end
+end
+
+# function map_interpolate!(cc_field, points, oc_field::OC.Field)
+#     loc = map(instantiate, OC.Fields.location(oc_field))
+#     grid = oc_field.grid
+#     data = oc_field.data
+
+#     map!(cc_field, points) do pt
+#         FT = eltype(pt)
+#         fᵢ = OC.Fields.interpolate(to_node(pt), data, loc, grid)
+#         convert(FT, fᵢ)
+#     end
+
+#     return nothing
+# end
+
+function surface_flux(f::OC.AbstractField)
+    top_bc = f.boundary_conditions.top
+    if top_bc isa OC.BoundaryCondition{<:OC.BoundaryConditions.Flux}
+        return top_bc.condition
+    else
+        return nothing
+    end
+end
+
+function Interfacer.remap(field::OC.Field, target_space)
+    return map_interpolate(CC.Fields.coordinate_field(target_space), field)
+end
+
+function Interfacer.remap(operation::OC.AbstractOperations.AbstractOperation, target_space)
+    evaluated_field = OC.Field(operation)
+    OC.compute!(evaluated_field)
+    return Interfacer.remap(evaluated_field, target_space)
+end
+
+Interfacer.get_field(sim::OceananigansSimulation, ::Val{:area_fraction}) = sim.area_fraction
+
+# TODO: Better values for this
```

Comment:
these should also get the float type of the simulation instead of hardcoding Float32